deploy/roles/mig-ui/tasks/main.yml

```diff
@@ -12,7 +12,7 @@
   k8s:
     state: '{{ state }}'
     definition: "{{ lookup('template', 'mig-ui.namespace.yaml') | from_yaml }}"
-  when: mig_ui_create_namespace
+  when: mig_ui_remove_namespace or state == 'present'
```

Comment:
I don't think this logic is semantically correct. It probably works as intended, but it's going to run this task **always** when `mig_ui_remove_namespace` is true, which isn't really what we want.

We want to "delete this namespace when state absent and the user has specifically indicated they want to remove the namespace, OR when the state is present to ensure the namespace exists. I would change this to:

`(mig_ui_remove_namespace and state == 'absent') or state == 'present'`